soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -32,6 +32,9 @@
 #include "usb.h"
 #endif
 
+/* System clock frequency */
+extern uint32_t SystemCoreClock;
```

Comment:
it is not clear to me what you are using this for but perhaps `static uint32_t SystemCoreClock;` is better if you take Derek's suggestion to use this variable to reduce the code size

Reply:
FYI, `SystemCoreClock` is a global variable used by NXP's MCUXpresso SDK drivers for the system clock frequency.  It should be updated whenever the frequency of the main clock is changed.